packages/ra-ui-materialui/src/input/useSupportCreateSuggestion.tsx

```diff
@@ -62,7 +61,7 @@ export const useSupportCreateSuggestion = (
                             ? createHintValue
                             : createValue,
                 },
-                typeof optionText === 'string' ? optionText : 'name',
```

Comment:
I believe this was made like this for people providing their own `renderOption` in `AutocompleteInput`. Now they'll have to check if the item is the createItem like us

Reply:
Right! I tried to think about all cases that would prevent me from simplifying the code to using only the `name` field, but forgot this one!
I've reverted this change and made the `useChoices` hook compatible with all cases (hopefully).